dpe/src/x509.rs

```diff
@@ -823,6 +843,43 @@ impl X509CertWriter<'_> {
         Ok(bytes_written)
     }
 
+    /// Encode a KeyUsage extension
+    ///
+    /// https://datatracker.ietf.org/doc/html/rfc5280
+    fn encode_key_usage(&mut self) -> Result<usize, DpeErrorCode> {
+        let key_usage_size = Self::get_key_usage_size(/*tagged=*/ false)?;
+
+        // Encode Extension
+        let mut bytes_written = self.encode_byte(Self::SEQUENCE_TAG)?;
+        bytes_written += self.encode_size_field(key_usage_size)?;
+        bytes_written += self.encode_oid(Self::KEY_USAGE_OID)?;
+
+        bytes_written += self.encode_byte(Self::BOOL_TAG)?;
+        bytes_written += self.encode_size_field(Self::BOOL_SIZE)?;
+        bytes_written += self.encode_byte(0xFF)?;
+
+        // Extension data is sequence -> octet string. To compute size, wrap
+        // in tagging twice.
+        bytes_written += self.encode_byte(Self::OCTET_STRING_TAG)?;
+        bytes_written += self.encode_size_field(Self::get_structure_size(
+            Self::get_structure_size(1, /*tagged=*/ true)?,
+            /*tagged=*/ true,
+        )?)?;
+
+        bytes_written += self.encode_byte(Self::BIT_STRING_TAG)?;
+        bytes_written +=
+            self.encode_size_field(Self::get_structure_size(1, /*tagged=*/ true)?)?;
+        // First byte of BIT STRING is the number of unused bits. But all bits
+        // are used.
+        bytes_written += self.encode_byte(0)?;
+
+        // Set digitalSignature bit
+        bytes_written += self.encode_byte(0x80)?;
```

Comment:
In the x.509 spec, it looks like KeyUsage can be 9 bits. I'm not sure if some parsers will be unhappy if only 8 bits are found.

I didn't see anything in the spec about whether all bits need to be present or not.